### ISSUE TITLE
Loss explosion during training

### ISSUE THREAD

**Issue by shp216** [OmniSVG/OmniSVG#44]
<img width="447" height="237" alt="Image" src="https://github.com/user-attachments/assets/c9cbf205-1390-4a7c-9cb6-21e96a875d75" />

I’ve been fine-tuning Qwen-VL on an Icon dataset using the Qwen'sofficial repository. While the loss generally decreases well, I’ve consistently observed sudden spikes in gradient norms during training. When this happens, the loss jumps significantly and doesn't return to its previous level, although it starts decreasing again from that new, higher point. Have you experienced this kind of instability before? Any insights on how to mitigate these gradient spikes would be appreciated.

**Comment by YiyingYang12** [comment]
Hi @shp216,

The gradient spikes and loss explosion you're experiencing might be related to the training configuration or data preprocessing pipeline.

We recommend trying our official training codebase and pre-processed Icon dataset:

- **Training Code**: [OpenVGLab/OmniSVG-train](https://github.com/OpenVGLab/OmniSVG-train)
- **Pre-processed Icon Dataset**: [MMSVG-Icon on Hugging Face](https://huggingface.co/datasets/OmniSVG/MMSVG-Icon)

 Please give it a try and let us know if you still encounter similar issues.

Hope this helps!

**Comment by shp216** [comment]
Thank you so much for your kind support and thoughtfulness. I truly appreciate your help. Wishing you a wonderful and happy New Year!

**Comment by YiyingYang12** [comment]
You're very welcome, @shp216! Happy New Year to you too! 🎉

Please don't hesitate to reach out if you run into any other issues down the road. We're here to help!

**Comment by shp216** [comment]
Hi @YiyingYang12, Could you please let me know what value the training loss converged to? 
Thanks!
